sentry-core/src/main/java/io/sentry/core/SentryClient.java

```diff
@@ -63,14 +60,17 @@ public SentryClient(final @NotNull SentryOptions options, @Nullable Connection c
 
     options.getLogger().log(SentryLevel.DEBUG, "Capturing event: %s", event.getEventId());
 
+    boolean sendEvent = true;
+
     if (ApplyScopeUtils.shouldApplyScopeData(hint)) {
       // Event has already passed through here before it was cached
       // Going through again could be reading data that is no longer relevant
       // i.e proguard id, app version, threads
       event = applyScope(event, scope, hint);
 
       if (event == null) {
```

Comment:
Could the event be null after calling `applyScope`? 
Shouldn't we check at the start of the method, then only on beforeSend, and event processors?

Reply:
the event can be null when running the scope's processor